src/components/icon/icon.js

```diff
@@ -20,25 +21,55 @@ angular.module('material.components.icon', [
  * @restrict E
  *
  * @description
- * The `<md-icon>` directive is an element useful for SVG icons
+ * The `<md-icon>` directive is an element useful for showing an icon
  *
  * @usage
  * <hljs lang="html">
- *  <md-icon icon="/img/icons/ic_access_time_24px.svg">
- *  </md-icon>
+ *  <md-icon md-icon-key="icon-android" md-icon-size=""></md-icon>
  * </hljs>
  *
  */
-function mdIconDirective() {
+function mdIconDirective($mdIcon) {
   return {
+    scope: {
+      mdIconKey: '@',
+      mdIconSize: '@',
+      mdIconColor: '@'
+    },
     restrict: 'E',
-    template: '<object class="md-icon"></object>',
-    compile: function(element, attr) {
-      var object = angular.element(element[0].children[0]);
-      if(angular.isDefined(attr.icon)) {
-        object.attr('data', attr.icon);
+    replace: true,
+    template: '<span class="icon"></span>',
```

Comment:
This should probably have `role="img"` since it represents an image. It also needs a text alternative, either with a child node containing offscreen text or an `aria-label` attribute.